hnn_core/network_builder.py

```diff
@@ -415,17 +415,17 @@ def _create_cells_and_feeds(self, threshold, record_vsoma=False,
                 _PC.cell(drive_cell.gid, drive_cell.nrn_netcon)
                 self._drive_cells.append(drive_cell)
 
-    def _connect_celltypes(self, src_type, target_type, loc,
+    def _connect_celltypes(self, connectivity, loc,
                            receptor, nc_dict, unique=False,
                            allow_autapses=True):
         """Connect two cell types for a particular receptor.
 
         Parameters
         ----------
-        src_type : str
-            The source cell type
-        target_type : str
-            The target cell type
+        connectivity : list of list [[str, int, str, int], ...]
+            Outer list contains an element for each connection. The inner
+            list specifies connections between cells with the form:
+            [src_type, src_gid, target_type, target_gid]
```

Comment:
I would actually leave out `src_type` and `target_type` ... you can use the `gid_dict` to figure it out. There was a helper function IIRC and it was being used somewhere

Reply:
Or if you want to prevent the extra computation (is it much anyway?), I'd suggest a dict of list. So you have:

connectivity['src_gids']
connectivity['src_types']
connectivity['target_gids']
connectivity['target_types']